test/unit_tests/util/RAIIWrappersTest.cpp

```diff
@@ -0,0 +1,69 @@
+#include <gtest/gtest.h>
+
+#include "util/RAII_Wrappers.h"
+
+namespace {
+class TestData {
+public:
+    TestData() { nbData++; };
+    ~TestData() { nbData--; };
+
+    int value = 2;
+
+    int ref_count = 1;
+
+    static TestData* ref(TestData* p) {
+        p->ref_count++;
+        return p;
+    }
+    static void unref(TestData* p) {
+        p->ref_count--;
+        if (p->ref_count == 0) {
+            delete (p);
+        }
+    }
+
+    static int nbData;
+};
+int TestData::nbData = 0;
+};  // namespace
+
+using TestRAII = xoj::util::RAII_Wrapper<TestData, TestData::ref, TestData::unref>;
+
+TEST(UtilsRAII, testWrappers) {
```

Comment:
Please test only one thing per test. And then give the test a very descriptive name, please.